diff -Naur orig/cegui/src/XMLParserModules/TinyXMLParser/CEGUITinyXMLParser.cpp patched/cegui/src/XMLParserModules/TinyXMLParser/CEGUITinyXMLParser.cpp
--- orig/cegui/src/XMLParserModules/TinyXMLParser/CEGUITinyXMLParser.cpp	2013-01-12 18:07:24.134739579 +0100
+++ patched/cegui/src/XMLParserModules/TinyXMLParser/CEGUITinyXMLParser.cpp	2017-10-09 23:01:39.013979729 +0200
@@ -145,7 +145,7 @@
                 processElement(childNode->ToElement());
                 break;
             case CEGUI_TINYXML_NAMESPACE::TiXmlNode::CEGUI_TINYXML_TEXT:
-                if (childNode->ToText()->Value() != '\0')
+                if (childNode->ToText()->Value() != NULL)
                     d_handler->text((utf8*)childNode->ToText()->Value());
                 break;
 
