diff -Nur orig/Makefile mod/Makefile
--- orig/Makefile	2017-10-02 02:23:52.000000000 +0200
+++ mod/Makefile	2017-12-16 13:53:48.319145236 +0100
@@ -44,7 +44,7 @@
 
 # Everything gets put in subdirectories of. . .
 
-TOPDIR=		/usr/local
+TOPDIR=		/usr/
 
 # "Compiled" time zone information is placed in the "TZDIR" directory
 # (and subdirectories).
@@ -52,7 +52,7 @@
 # TZDIR_BASENAME should not contain "/" and should not be ".", ".." or empty.
 
 TZDIR_BASENAME=	zoneinfo
-TZDIR=		$(TOPDIR)/etc/$(TZDIR_BASENAME)
+TZDIR=		$(TOPDIR)/share/$(TZDIR_BASENAME)
 
 # Types to try, as an alternative to time_t.  int64_t should be first.
 TIME_T_ALTERNATIVES= int64_t int32_t uint32_t uint64_t
@@ -67,7 +67,7 @@
 
 # Manual pages go in subdirectories of. . .
 
-MANDIR=		$(TOPDIR)/man
+MANDIR=		$(TOPDIR)/share/man
 
 # Library functions are put in an archive in LIBDIR.
 
@@ -479,7 +479,7 @@
 			$(DESTDIR)$(MANDIR)/man8
 		$(ZIC_INSTALL) -l $(LOCALTIME) -p $(POSIXRULES)
 		cp -f $(TABDATA) $(DESTDIR)$(TZDIR)/.
-		cp tzselect zic zdump $(DESTDIR)$(ETCDIR)/.
+		cp tzselect zic zdump $(DESTDIR)$(BINDIR)/.
 		cp libtz.a $(DESTDIR)$(LIBDIR)/.
 		$(RANLIB) $(DESTDIR)$(LIBDIR)/libtz.a
 		cp -f newctime.3 newtzset.3 $(DESTDIR)$(MANDIR)/man3/.
