diff -Naur orig/data/nemo.desktop.in.in patched/data/nemo.desktop.in.in
--- orig/data/nemo.desktop.in.in	2017-08-08 13:15:08.134739579 +0200
+++ patched/data/nemo.desktop.in.in	2017-10-08 03:34:29.183046589 +0200
@@ -33,8 +33,8 @@
 Name[fr_CA]=Fichiers
 Name[fr_FR]=Fichiers
 Name[hr]=Nemo
-Name[hu]=Fájlok
-Name[hu_HU]=Fájlok
+Name[hu]=Nemo
+Name[hu_HU]=Nemo
 Name[ko]=파일
 Name[ko_KR]=파일
 Name[lt]=Failai
