diff -Nur orig/lib/Makefile.am mod/lib/Makefile.am
--- orig/lib/Makefile.am	2017-10-06 14:40:42.000000000 +0200
+++ mod/lib/Makefile.am	2017-12-20 08:29:47.524449305 +0100
@@ -5,8 +5,7 @@
 
 lib_LTLIBRARIES = libzip.la
 noinst_HEADERS = zipint.h gladman-fcrypt.h
-include_HEADERS = zip.h
-nodist_include_HEADERS = zipconf.h
+include_HEADERS = zip.h zipconf.h
 
 # also update CMakeLists.txt when changing version
 libzip_la_LDFLAGS=-no-undefined -version-info 5:0:0
