diff -Nur orig/libzip.pc.in mod/libzip.pc.in
--- orig/libzip.pc.in	2017-03-24 10:22:29.000000000 +0100
+++ mod/libzip.pc.in	2017-09-18 09:53:33.417112959 +0200
@@ -2,7 +2,6 @@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
-libincludedir=@libdir@/@PACKAGE@/include
 
 zipcmp=@prefix@/bin/zipcmp
 
@@ -10,5 +9,5 @@
 Description: library for handling zip archives
 Version: @VERSION@
 Libs: @PKG_CONFIG_RPATH@ -L${libdir} -lzip @LIBS@
-Cflags: -I${includedir} -I${libincludedir}
+
 
