diff -Nur orig/scripts/blk_availability_systemd_red_hat.service.in mod/scripts/blk_availability_systemd_red_hat.service.in
--- orig/scripts/blk_availability_systemd_red_hat.service.in	2017-07-20 18:15:30.000000000 +0200
+++ mod/scripts/blk_availability_systemd_red_hat.service.in	2017-08-31 17:29:47.378711497 +0200
@@ -6,7 +6,7 @@
 
 [Service]
 Type=oneshot
-ExecStart=/usr/bin/true
+ExecStart=/bin/true
 ExecStop=@sbindir@/blkdeactivate -u -l wholevg -m disablequeueing
 RemainAfterExit=yes
 
