diff -Nur orig/configure mod/configure
--- orig/configure	2015-12-03 09:29:54.000000000 +0100
+++ mod/configure	2017-10-17 17:43:46.000000000 +0200
@@ -4667,9 +4667,9 @@
   ## First, let's just see if we can find Guile at all.
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for Guile" >&5
 $as_echo_n "checking for Guile... " >&6; }
-  GUILE_BIN="guile18"
-  GUILE_CONFIG="guile18-config"
-  guile18-config link > /dev/null || {
+  GUILE_BIN="guile1.8"
+  GUILE_CONFIG="guile-config1.8"
+  guile-config1.8 link > /dev/null || {
     GUILE_BIN="guile"
     GUILE_CONFIG="guile-config"
     guile-config link > /dev/null || {
diff -Nur orig/src/Scheme/Guile/guile_tm.hpp mod/src/Scheme/Guile/guile_tm.hpp
--- orig/src/Scheme/Guile/guile_tm.hpp	2015-08-22 11:54:52.000000000 +0200
+++ mod/src/Scheme/Guile/guile_tm.hpp	2017-10-17 17:41:50.000000000 +0200
@@ -38,7 +38,7 @@
 
 #if defined(GUILE_D) || defined(GUILE_C) 
 #ifdef GUILE_BIN_guile18
-#include <libguile18.h>
+#include <guile/gh.h>
 #else
 #include <libguile.h>
 #endif
