diff -Naur util-linux-2.30.1-orig/misc-utils/whereis.c util-linux-2.30.1/misc-utils/whereis.c
--- util-linux-2.30.1-orig/misc-utils/whereis.c	2017-07-20 10:16:26.354984891 +0200
+++ util-linux-2.30.1/misc-utils/whereis.c	2017-08-22 22:51:35.002551095 +0200
@@ -96,75 +96,21 @@
 static const char *bindirs[] = {
 	"/usr/bin",
 	"/usr/sbin",
-	"/usr/lib",
-	"/usr/lib64",
 	"/bin",
 	"/sbin",
-	"/etc",
-	"/usr/etc",
-	"/lib",
-	"/lib64",
-	"/usr/games",
-	"/usr/games/bin",
-	"/usr/games/lib",
-	"/usr/emacs/etc",
-	"/usr/lib/emacs/*/etc",
-	"/usr/TeX/bin",
-	"/usr/tex/bin",
-	"/usr/interviews/bin/LINUX",
-
-	"/usr/X11R6/bin",
-	"/usr/X386/bin",
-	"/usr/bin/X11",
-	"/usr/X11/bin",
-	"/usr/X11R5/bin",
-
 	"/usr/local/bin",
 	"/usr/local/sbin",
-	"/usr/local/etc",
-	"/usr/local/lib",
-	"/usr/local/games",
-	"/usr/local/games/bin",
-	"/usr/local/emacs/etc",
-	"/usr/local/TeX/bin",
-	"/usr/local/tex/bin",
-	"/usr/local/bin/X11",
-
-	"/usr/contrib",
-	"/usr/hosts",
-	"/usr/include",
-
-	"/usr/g++-include",
-
-	"/usr/ucb",
-	"/usr/old",
-	"/usr/new",
-	"/usr/local",
-	"/usr/libexec",
-	"/usr/share",
-
 	"/opt/*/bin",
 	NULL
 };
 
 static const char *mandirs[] = {
-	"/usr/man/*",
 	"/usr/share/man/*",
-	"/usr/X386/man/*",
-	"/usr/X11/man/*",
-	"/usr/TeX/man/*",
-	"/usr/interviews/man/mann",
-	"/usr/share/info",
 	NULL
 };
 
 static const char *srcdirs[] = {
 	"/usr/src/*",
-	"/usr/src/lib/libc/*",
-	"/usr/src/lib/libc/net/*",
-	"/usr/src/ucb/pascal",
-	"/usr/src/ucb/pascal/utilities",
-	"/usr/src/undoc",
 	NULL
 };
 
