diff -Nur orig/profile/flatpak.sh mod/profile/flatpak.sh
--- orig/profile/flatpak.sh	2019-09-10 12:47:22.000000000 +0200
+++ mod/profile/flatpak.sh	2019-09-10 13:04:24.870262928 +0200
@@ -1,15 +1,23 @@
 # set XDG_DATA_DIRS to include Flatpak installations
 
-new_dirs=
-while read -r install_path
-do
-    share_path=$install_path/exports/share
-    case ":$XDG_DATA_DIRS:" in
-        *":$share_path:"*) :;;
-        *":$share_path/:"*) :;;
-        *) new_dirs=${new_dirs:+${new_dirs}:}$share_path;;
-    esac
-done < <(echo "${XDG_DATA_HOME:-"$HOME/.local/share"}/flatpak"; flatpak --installations)
+new_dirs=$(
+    (
+        echo "${XDG_DATA_HOME:-"$HOME/.local/share"}/flatpak"
+        flatpak --installations
+    ) | (
+        new_dirs=
+        while read -r install_path
+        do
+            share_path=$install_path/exports/share
+            case ":$XDG_DATA_DIRS:" in
+                *":$share_path:"*) :;;
+                *":$share_path/:"*) :;;
+                *) new_dirs=${new_dirs:+${new_dirs}:}$share_path;;
+            esac
+        done
+        echo "$new_dirs"
+    )
+)
 
 export XDG_DATA_DIRS
 XDG_DATA_DIRS="${new_dirs:+${new_dirs}:}${XDG_DATA_DIRS:-/usr/local/share:/usr/share}"
