diff -Nur orig/scite/gtk/SciTE.desktop mod/scite/gtk/SciTE.desktop
--- orig/scite/gtk/SciTE.desktop	2017-08-16 09:34:20.000000000 +0200
+++ mod/scite/gtk/SciTE.desktop	2017-10-17 16:29:59.841411666 +0200
@@ -1,6 +1,6 @@
 [Desktop Entry]
-Version=1.0
 Name=SciTE Text Editor
+Name[hu]=SciTE szövegszerkesztő
 Name[es]=Editor de textos SciTE
 Name[fr]=Éditeur de texte SciTE
 Name[ru]=Текстовый редактор SciTE
@@ -8,6 +8,8 @@
 Comment[es]=Edita especializado para programación
 Comment[fr]=Éditer des fichiers sources
 Comment[ru]=Редактор исходных кодов
+Comment[hu]=SciTE fejlesztői szövegszerkesztő
+GenericName[hu]=Szövegszerkesztő
 GenericName=Text Editor
 Type=Application
 Exec=SciTE %F
