From d8960ea9097e5a324acd04ce3f03f2d8bfe88df6 Mon Sep 17 00:00:00 2001
From: Rex Dieter <rdieter@gmail.com>
Date: Tue, 31 Jul 2018 09:45:37 -0500
Subject: Fix build against Qt 5.11+

---
 Apper/Settings/Settings.cpp | 1 +
 libapper/Requirements.h     | 1 +
 2 files changed, 2 insertions(+)

diff --git a/Apper/Settings/Settings.cpp b/Apper/Settings/Settings.cpp
index e077b8e..615d86d 100644
--- a/Apper/Settings/Settings.cpp
+++ b/Apper/Settings/Settings.cpp
@@ -27,6 +27,7 @@
 #include <Enum.h>
 #include <PkStrings.h>
 
+#include <QAction>
 #include <QTimer>
 #include <QSortFilterProxyModel>
 
diff --git a/libapper/Requirements.h b/libapper/Requirements.h
index 730644b..8325326 100644
--- a/libapper/Requirements.h
+++ b/libapper/Requirements.h
@@ -23,6 +23,7 @@
 
 #include <QDialog>
 
+#include <QButtonGroup>
 #include <QToolButton>
 
 namespace Ui {
-- 
cgit v1.1

