# Fixed a compiler warning in ExchangeChars().
#
--- a/recording.c	2019/03/19 15:56:58	5.2
+++ b/recording.c	2019/05/06 11:26:06
@@ -610,7 +610,7 @@
                              char buf[4];
                              sprintf(buf, "#%02X", (unsigned char)*p);
                              memmove(p + 2, p, strlen(p) + 1);
-                             strncpy(p, buf, 3);
+                             memcpy(p, buf, 3);
                              p += 2;
                              }
                           else
