# Fixed handling PATs that contain no PMTs.
#
--- a/pat.c	2016/12/23 14:02:07	5.0
+++ b/pat.c	2019/03/11 13:20:27
@@ -372,7 +372,8 @@
                }
            if (numPmtEntries > 0 && pmtIndex < 0)
               pmtIndex = 0;
-           Add(GetPmtPid(pmtIndex), SI::TableIdPMT);
+           if (pmtIndex >= 0)
+              Add(GetPmtPid(pmtIndex), SI::TableIdPMT);
            patVersion = pat.getVersionNumber();
            timer.Set(PMT_SCAN_TIMEOUT);
            }
