# Fixed handling repeat function for keyboards.
#
--- a/remote.c	2015/01/20 14:53:57	5.0
+++ b/remote.c	2019/05/13 13:01:44
@@ -307,7 +307,7 @@
 int cKbdRemote::ReadKey(void)
 {
   cPoller Poller(STDIN_FILENO);
-  if (Poller.Poll(50)) {
+  if (Poller.Poll(Setup.RcRepeatDelta * 3 / 2)) {
      uchar ch = 0;
      int r = safe_read(STDIN_FILENO, &ch, 1);
      if (r == 1)
@@ -431,6 +431,11 @@
            FirstCommand = 0;
            FirstTime.Set();
            }
+        else if (FirstCommand && FirstTime.Elapsed() > (uint)Setup.RcRepeatDelay) {
+           // Don't wait too long for that second key press:
+           Delayed = false;
+           FirstCommand = 0;
+           }
         LastCommand = Command;
         }
 }
