diff -Nur orig/settings.conf mod/settings.conf
--- orig/settings.conf	2019-08-07 12:58:22.000000000 +0200
+++ mod/settings.conf	2019-09-07 15:39:29.483937053 +0200
@@ -93,6 +93,7 @@
   - partition
   - users
 #  - tracking
+  - netinstall
   - summary
 - exec:
 #  - dummycpp
@@ -101,25 +102,21 @@
 #  - dummypythonqt
   - partition
   - mount
-  - unpackfs
+  - chrootcfg
+  - networkcfg
+  - packages
   - machineid
   - fstab
   - locale
-  - keyboard
   - localecfg
 #  - luksbootkeyfile
 #  - luksopenswaphookcfg
 #  - dracutlukscfg
 #  - plymouthcfg
-  - initcpiocfg
-  - initcpio
   - users
   - displaymanager
-  - networkcfg
   - hwclock
-  - services-systemd
 #  - dracut
-  - initramfs
 #  - grubcfg
   - bootloader
   - umount
@@ -164,6 +161,10 @@
 # YAML: boolean.
 dont-chroot: false
 
+packageOperations:
+  -install
+  -remove
+
 # If this is set to true, Calamares refers to itself as a "setup program"
 # rather than an "installer". Defaults to the value of dont-chroot, but
 # Calamares will complain if this is not explicitly set.
