diff -Nur orig/settings.conf mod/settings.conf
--- orig/settings.conf	2019-08-07 12:58:22.000000000 +0200
+++ mod/settings.conf	2019-09-19 12:36:28.699121139 +0200
@@ -87,40 +87,27 @@
 sequence:
 - show:
   - welcome
-#  - dummypythonqt
   - locale
   - keyboard
   - partition
   - users
-#  - tracking
+  - netinstall
   - summary
 - exec:
-#  - dummycpp
-#  - dummyprocess
-#  - dummypython
-#  - dummypythonqt
   - partition
   - mount
-  - unpackfs
+  - chrootcfg
+  - networkcfg
+  - packages
   - machineid
   - fstab
   - locale
-  - keyboard
   - localecfg
-#  - luksbootkeyfile
-#  - luksopenswaphookcfg
-#  - dracutlukscfg
-#  - plymouthcfg
-  - initcpiocfg
-  - initcpio
   - users
   - displaymanager
-  - networkcfg
   - hwclock
-  - services-systemd
-#  - dracut
-  - initramfs
-#  - grubcfg
+  - dracut
+  - grubcfg
   - bootloader
   - umount
 - show:
@@ -164,6 +151,10 @@
 # YAML: boolean.
 dont-chroot: false
 
+packageOperations:
+  -install
+  -remove
+
 # If this is set to true, Calamares refers to itself as a "setup program"
 # rather than an "installer". Defaults to the value of dont-chroot, but
 # Calamares will complain if this is not explicitly set.
