diff -Nur orig/CMakeLists.txt mod/CMakeLists.txt
--- orig/CMakeLists.txt	2017-12-05 17:00:16.000000000 +0100
+++ mod/CMakeLists.txt	2018-06-19 12:44:35.822802238 +0200
@@ -142,7 +142,7 @@
         string(REGEX REPLACE "[ \t\n]+" "" SYSTEMD_SYSTEM_UNIT_DIR ${SYSTEMD_SYSTEM_UNIT_DIR})
     endif()
 
-    set(MINIMUM_VT 1)
+    set(MINIMUM_VT 7)
     set(HALT_COMMAND "/usr/bin/systemctl poweroff")
     set(REBOOT_COMMAND "/usr/bin/systemctl reboot")
 else()
diff -Nur orig/services/sddm.service.in mod/services/sddm.service.in
--- orig/services/sddm.service.in	2017-12-05 17:00:16.000000000 +0100
+++ mod/services/sddm.service.in	2018-06-19 12:45:46.286682825 +0200
@@ -1,11 +1,13 @@
 [Unit]
 Description=Simple Desktop Display Manager
 Documentation=man:sddm(1) man:sddm.conf(5)
-Conflicts=getty@tty1.service
-After=systemd-user-sessions.service getty@tty1.service plymouth-quit.service
+Conflicts=getty@tty7.service
+After=systemd-user-sessions.service getty@tty1.service plymouth-quit.service systemd-logind.service
 
 [Service]
+
 ExecStart=@CMAKE_INSTALL_FULL_BINDIR@/sddm
+ExecStartPre=/bin/sleep 5
 Restart=always
 
 [Install]
