diff -Nur orig/src/gst/parole-gst.c mod/src/gst/parole-gst.c
--- orig/src/gst/parole-gst.c	2018-03-01 04:00:32.000000000 +0100
+++ mod/src/gst/parole-gst.c	2018-10-20 12:17:07.047183118 +0200
@@ -1411,33 +1411,6 @@
             parole_gst_query_duration(gst);
         }
         break;
-    case GST_MESSAGE_ELEMENT:
-        if (gst_is_missing_plugin_message(msg)) {
-            g_debug("Missing plugin\n");
-            parole_gst_stop(gst);
-            dialog = parole_gst_missing_codec_dialog(gst, msg);
-            response = gtk_dialog_run(dialog);
-            if ( response == GTK_RESPONSE_ACCEPT ) {
-                 gtk_widget_destroy(GTK_WIDGET(dialog));
-                 details[0] = gst_missing_plugin_message_get_installer_detail(msg);
-                 details[1] = NULL;
-                 ctx = gst_install_plugins_context_new();
-
-#ifdef GDK_WINDOWING_X11
-                if (gtk_widget_get_window(GTK_WIDGET(gst)) != NULL && gtk_widget_get_realized(GTK_WIDGET(gst))) {
-                    gst_install_plugins_context_set_xid(ctx,
-                        gdk_x11_window_get_xid(gtk_widget_get_window(GTK_WIDGET(gst))));
-                }
-#endif /* GDK_WINDOWING_X11 */
-
-                 gst_install_plugins_async((const gchar * const *) details, ctx,
-                                           parole_gst_install_plugins_result_func, gst);
-                 gst_install_plugins_context_free(ctx);
-            } else if ( (response == GTK_RESPONSE_REJECT) || (response == GTK_RESPONSE_OK) ) {
-                gtk_widget_destroy(GTK_WIDGET(dialog));
-            }
-        }
-        break;
     case GST_MESSAGE_INFO:
         TRACE("Info message:");
         break;
