diff -Nur orig/configure.ac mod/configure.ac
--- orig/configure.ac	2017-10-01 22:39:02.000000000 +0200
+++ mod/configure.ac	2017-10-04 20:13:19.580250992 +0200
@@ -64,7 +64,7 @@
 WEBKITGTK_REQUIRED_VERSION=2.10.0
 
 
-PKG_CHECK_MODULES([TRACKER], [tracker-sparql-1.0], [sparql_version="tracker-sparql-1.0"],
+PKG_CHECK_MODULES([TRACKER], [tracker-sparql-2.0], [sparql_version="tracker-sparql-2.0"],
    [PKG_CHECK_MODULES([TRACKER], [tracker-sparql-0.18], [sparql_version="tracker-sparql-0.18"])
 ])
 
