diff -Nur orig/projects/compiler-rt/lib/esan/esan_sideline_linux.cpp mod/projects/compiler-rt/lib/esan/esan_sideline_linux.cpp
--- orig/projects/compiler-rt/lib/esan/esan_sideline_linux.cpp	2016-07-19 07:03:38.000000000 +0200
+++ mod/projects/compiler-rt/lib/esan/esan_sideline_linux.cpp	2017-08-27 19:21:58.327642659 +0200
@@ -70,7 +70,7 @@
 
   // Set up a signal handler on an alternate stack for safety.
   InternalScopedBuffer<char> StackMap(SigAltStackSize);
-  struct sigaltstack SigAltStack;
+  stack_t SigAltStack;
   SigAltStack.ss_sp = StackMap.data();
   SigAltStack.ss_size = SigAltStackSize;
   SigAltStack.ss_flags = 0;
