diff -Naur orig/src/basic/user-util.c patched/src/basic/user-util.c
--- orig/src/basic/user-util.c	2016-07-25 21:49:47.134739579 +0200
+++ patched/src/basic/user-util.c	2017-07-08 01:29:25.840159971 +0200
@@ -133,6 +133,9 @@
 
         if (parse_uid(*username, &u) >= 0) {
                 errno = 0;
+                p = getpwnam(*username);
+        } else {
+                errno = 0;
                 p = getpwuid(u);
 
                 /* If there are multiple users with the same id, make
@@ -142,9 +145,6 @@
                  * pick the real username from /etc/passwd. */
                 if (p)
                         *username = p->pw_name;
-        } else {
-                errno = 0;
-                p = getpwnam(*username);
         }
 
         if (!p)
