diff -Nur orig/setup.py mod/setup.py
--- orig/setup.py	2015-02-19 14:33:25.000000000 +0100
+++ mod/setup.py	2018-01-11 10:15:15.270394021 +0100
@@ -241,7 +241,7 @@
 files = make_source_list(paxtkinter_src, ['paxtkinter.c'])
 paxtkinter_module = Extension('app.paxtkinter',
 			include_dirs=tcl_include_dirs,
-			libraries=['tk' + tcl_ver, 'tcl' + tcl_ver],
+			libraries=['tk' + '8.7', 'tcl' + tcl_ver],
 			define_macros=macros, sources=files)
 modules.append(paxtkinter_module)
 
@@ -274,7 +274,7 @@
 
 pax_src = os.path.join(src_path, 'extensions', 'pax')
 pax_include_dirs = ['/usr/include/cairo'] + tcl_include_dirs
-pax_libs = ['X11', 'Xext', 'tk' + tcl_ver, 'tcl' + tcl_ver, 'cairo']
+pax_libs = ['X11', 'Xext', 'tk' + '8.7', 'tcl' + tcl_ver, 'cairo']
 files = ['borderobject.c', 'clipmask.c', 'cmapobject.c', 'fontobject.c',
 	'gcobject.c', 'imageobject.c', 'intl.c', 'paxmodule.c', 'paxutil.c',
 	'pixmapobject.c', 'regionobject.c', 'tkwinobject.c']
@@ -300,7 +300,7 @@
 files = make_source_list(tkpng_src, ['tkImgPNG.c', 'tkImgPNGInit.c'])
 tkpng_module = Extension('sk1sdk.tkpng.libtkpng',
 			include_dirs=tcl_include_dirs,
-			libraries=['tk' + tcl_ver, 'tcl' + tcl_ver, 'z'],
+			libraries=['tk' + '8.7', 'tcl' + tcl_ver, 'z'],
 			define_macros=macros, sources=files)
 modules.append(tkpng_module)
 
@@ -308,7 +308,7 @@
 files = make_source_list(tkXcursor_src, ['_tkXcursor.c'])
 tkXcursor_module = Extension('sk1sdk.tkXcursor._tkXcursor',
 			include_dirs=['/usr/include/X11/Xcursor'] + tcl_include_dirs,
-			libraries=['tk' + tcl_ver, 'tcl' + tcl_ver, 'Xcursor'],
+			libraries=['tk' + '8.7', 'tcl' + tcl_ver, 'Xcursor'],
 			define_macros=macros, sources=files)
 modules.append(tkXcursor_module)
 
@@ -316,7 +316,7 @@
 files = make_source_list(tkcairo_src, ['_tkcairo.c'])
 tkcairo_module = Extension('sk1sdk.tkcairo._tkcairo',
 			include_dirs=['/usr/include/cairo'] + tcl_include_dirs,
-			libraries=['tk' + tcl_ver, 'tcl' + tcl_ver, 'cairo'],
+			libraries=['tk' + '8.7', 'tcl' + tcl_ver, 'cairo'],
 			define_macros=macros, sources=files)
 modules.append(tkcairo_module)
 
