Author: Francois Marier <francois@debian.org>
Last-Update: 2016-09-17
Description: Mark applet as compatible with gnome-shell 3.22
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=837214
Forwarded: yes, https://github.com/rcaelers/workrave/pull/80

--- a/frontend/applets/gnome-shell/src/metadata.json.in
+++ b/frontend/applets/gnome-shell/src/metadata.json.in
@@ -3,7 +3,7 @@
  "uuid": "@uuid@",
  "name": "Workrave",
  "description": "Applet that shows all the Workrave timers.",
- "shell-version": [ "3.2", "3.3", "3.4", "3.5", "3.5.4", "3.6.0", "3.6", "3.8", "3.10", "3.12", "3.14", "3.16", "3.18", "3.20", "3.22", "@shell_current@" ],
+ "shell-version": [ "3.2", "3.3", "3.4", "3.5", "3.5.4", "3.6.0", "3.6", "3.8", "3.10", "3.12", "3.14", "3.16", "3.18", "3.20", "3.21.91", "3.21.92", "3.22", "@shell_current@" ],
  "localedir": "@LOCALEDIR@",
  "url": "@url@"
 }
